src/physics/article.js

```diff
@@ -443,19 +448,61 @@ export function executeArticleHits (input){
 }
 
 export function wallDetection (i){
-    for (var j = 0; j < activeStage.wallL.length; j++) {
-        if (aArticles[i][2].ecb[1].y < activeStage.wallL[j][0].y && aArticles[i][2].ecb[1].y > activeStage.wallL[j][1].y && aArticles[
-                i][2].ecb[1].x >= activeStage.wallL[j][1].x && aArticles[i][2].ecb[1].x < activeStage.wallL[j][1].x) {
-            return true;
-        }
+  const article = aArticles[i][2];
+  const ecbp = article.ecb;
+  let ecb1;
+  if (article.timer < 2) {
+    const focus = article.posPrev;
+    const offset = 0.0001;
+    ecb1 = [ new Vec2D ( focus.x         , focus.y - offset ) 
+           , new Vec2D ( focus.x + offset, focus.y          )
+           , new Vec2D ( focus.x         , focus.y + offset )
+           , new Vec2D ( focus.x - offset, focus.y          )
+           ];
+  }
+  else {
+    ecb1 = moveECB(ecbp, subtract(article.posPrev,article.pos));
+  }
+  let collisions = [];
+  let thisCollision = null;
+  for (var j = 0; j < activeStage.wallL.length; j++) {
+    thisCollision = findCollision (ecb1, ecbp, [activeStage.wallL[j], ["l", j]] );
```

Comment:
Using the environmental collision routine to check for collision, as the articles have their own ECB.